src/main/java/it/gov/pagopa/payment/dto/mapper/TransactionInProgress2TransactionResponseMapper.java

```diff
@@ -2,19 +2,21 @@
 
 import it.gov.pagopa.payment.dto.qrcode.TransactionResponse;
 import it.gov.pagopa.payment.model.TransactionInProgress;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.function.TriFunction;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
-
-import java.util.function.Function;
+import org.springframework.web.util.UriComponentsBuilder;
 
 @Service
+@Slf4j
 public class TransactionInProgress2TransactionResponseMapper
-    implements Function<TransactionInProgress, TransactionResponse> {
+    implements TriFunction<TransactionInProgress, String, String, TransactionResponse> {
 
   @Value("${app.qrCode.expirations.authorizationMinutes}") int authorizationExpirationMinutes;
 
   @Override
-  public TransactionResponse apply(TransactionInProgress transactionInProgress) {
+  public TransactionResponse apply(TransactionInProgress transactionInProgress, String qrcodeImgBaseUrl, String qrcodeTxtBaseUrl) {
```

Comment:
let's add inside this Service the injection of qrcodeImgBaseUrl and qrcodeTxtBaseUrl